spec-tests/src/test/java/org/candlepin/spec/content/ContentAccessSpecTest.java

```diff
@@ -797,14 +810,77 @@ public void shouldReturnANotModifiedReturnCodeWhenTheDataHasNotBeenUpdatedSinceD
+        String ownerKey = owner.getKey();
+
+        ProductDTO prod = adminClient.ownerProducts().createProduct(ownerKey, Products.random());
+        ContentDTO content = adminClient.ownerContent().createContent(ownerKey, Contents.random());
+        adminClient.ownerProducts()
+            .addContentToProduct(ownerKey, prod.getId(), content.getId(), true);
+        adminClient.owners().createPool(ownerKey, Pools.random(prod));
+
+        ConsumerDTO consumer1 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer2 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer3 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer4 = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer5 = adminClient.consumers().createConsumer(Consumers.random(owner));
+
+        List<ConsumerDTO> consumers = List.of(consumer1, consumer2, consumer3, consumer4, consumer5);
+
+        int maxNumberOfAttempts = 5;
+        ApiException exception = assertThrows(ApiException.class, () -> {
+            for (int i = 0; i < maxNumberOfAttempts; i++) {
+                // Update the arch of all the consumers so that the content access payload key will be
+                // different than the key used to store the content access payload previously. This causes the
+                // regeneration of the content access payload which is what we need to cause the 503 response.
+                String newArch = StringUtil.random("arch-");
+                for (ConsumerDTO consumer : consumers) {
+                    consumer.putFactsItem(Facts.Arch.key(), newArch);
+                    adminClient.consumers().updateConsumer(consumer.getUuid(), consumer);
+                }
+
+                List<Callable<String>> tasks = List.of(
+                    () -> adminClient.consumers().getContentAccessBody(consumer1.getUuid(), null),
+                    () -> adminClient.consumers().getContentAccessBody(consumer2.getUuid(), null),
+                    () -> adminClient.consumers().getContentAccessBody(consumer3.getUuid(), null),
+                    () -> adminClient.consumers().getContentAccessBody(consumer4.getUuid(), null),
+                    () -> adminClient.consumers().getContentAccessBody(consumer5.getUuid(), null));
+
+                ExecutorService execService = Executors.newFixedThreadPool(consumers.size());
+                List<Future<String>> futures = execService.invokeAll(tasks);
+                execService.shutdown();
+                execService.awaitTermination(10L, TimeUnit.SECONDS);
+
+                for (Future<String> future : futures) {
+                    try {
+                        future.get();
+                    }
+                    catch (ExecutionException e) {
+                        throw (ApiException) e.getCause();
```

Comment:
This is a tad risky if anything else goes wrong in here -- any other exception will result in a `ClassCastException` when we try to rethrow. Brain is fairly toasted today, so the only thing that comes to mind is a simple `instanceof` check and rethrowing in both cases to ensure we fail the right way no matter what:

```
Throwable cause = e.getCause();
if (cause instanceof ApiException) {
  throw (ApiException) cause;
}

throw cause;
```

If there's a cleaner way to do this check and cast safely, feel free to use that instead.

Reply:
Updated

Reply:
Looking at this a little bit closer, we can just throw the `ExecutionException.getCause()` because the `assertThrows` will do the `instanceof` check and cast.